lib/chore/strategies/consumer/batcher.rb

```diff
@@ -11,27 +11,27 @@ def initialize(size)
         @size = size
         @batch = []
         @mutex = Mutex.new
-        @last_message = nil
         @callback = nil
         @running = true
       end
 
       # The main entry point of the Batcher, <tt>schedule</tt> begins a thread with the provided +batch_timeout+ 
       # as the only argument. While the Batcher is running, it will attempt to check if either the batch is full, 
-      # or if the +batch_timeout+ has elapsed since the last batch was executed. If the batch is full, it will be executed.
-      # If the +batch_timeout+ has elapsed, as soon as the next message enters the batch, it will be executed.
+      # or if the +batch_timeout+ has elapsed since the oldest message was added. If either case is true, the
+      # items in the batch will be executed.
       # 
       # Calling <tt>stop</tt> will cause the thread to finish it's current check, and exit
-      def schedule(batch_timeout=20)
+      def schedule(batch_timeout)
         @thread = Thread.new(batch_timeout) do |timeout|
           Chore.logger.info "Batching timeout thread starting"
           while @running do
             begin 
-              Chore.logger.debug "Last message added to batch: #{@last_message}: #{@batch.size}"
-              if @last_message && Time.now > (@last_message + timeout)
-                Chore.logger.debug "Batching timeout reached (#{@last_message + timeout}), current size: #{@batch.size}"
+              oldest_item = @batch.first
+              timestamp = oldest_item && oldest_item.created_at
+              Chore.logger.debug "Oldest message in batch: #{timestamp}, size: #{@batch.size}"
+              if timestamp && Time.now > (timestamp + timeout)
+                Chore.logger.debug "Batching timeout reached (#{timestamp + timeout}), current size: #{@batch.size}"
```

Comment:
I don't want that :) I'm happy to have you leave it as is.

On Wed, Mar 23, 2016 at 12:23 PM, Aaron Pfeifer <notifications@github.com>
wrote:

> In lib/chore/strategies/consumer/batcher.rb
> <https://github.com/Tapjoy/chore/pull/30#discussion_r57190195>:
>
> >          @thread = Thread.new(batch_timeout) do |timeout|
> >            Chore.logger.info "Batching timeout thread starting"
> >            while @running do
> >              begin
> > -              Chore.logger.debug "Last message added to batch: #{@last_message}: #{@batch.size}"
> > -              if @last_message && Time.now > (@last_message + timeout)
> > -                Chore.logger.debug "Batching timeout reached (#{@last_message + timeout}), current size: #{@batch.size}"
> > +              oldest_item = @batch.first
> > +              timestamp = oldest_item && oldest_item.created_at
> > +              Chore.logger.debug "Oldest message in batch: #{timestamp}, size: #{@batch.size}"
> > +              if timestamp && Time.now > (timestamp + timeout)
> > +                Chore.logger.debug "Batching timeout reached (#{timestamp + timeout}), current size: #{@batch.size}"
>
> Understand that it's not a blocker, but I'm not just going to ignore the
> feedback.
>
> In my opinion, inline assignment makes this less readable / clean because
> of the complexity of multiple conditions on the same line. I rarely see
> that style used in that scenario.
>
> The typical usage of inline assignment is:
>
> if a = my_complex_thing
>   do something with a
> end
>
> Not sure you want this PR to become a debate about inline assignment. If
> that's the style you prefer for chore, I'll update this -- otherwise not
> sure your feedback is actionable.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Tapjoy/chore/pull/30/files/51aa3c1b29c1a0d00b19bbe6109975fb43df4b00#r57190195>
>